yetanotherco/aligned_layer
# feat: add proving systems to help

## Description

- This PR adds the supported proving systems to the `aligned submit --help` command.
- In the process, I realized that we were using a String for this argument, and could be replaced for a ProvingSystemArg that maps to a ProvingSystemId. Due to this, we can delete the `parse_proving_systems` function.

## To Test

- Run `make install_aligned_compiling`.
- Run `which aligned` and make sure you are using the aligned client in `.cargo/bin/aligned`.
- Run `aligned submit --help` and you should see:
```
...
      --proving_system <Proving system>
          [possible values: GnarkPlonkBls12_381, GnarkPlonkBn254, Groth16Bn254, SP1, Halo2KZG, Halo2IPA]
...
```
- Send some proofs to check that the `--proving_system` arg is working correctly.

Closes #470 and #475.
